libs/core/langchain_core/messages/utils.py

```diff
@@ -384,36 +387,64 @@ def convert_to_messages(
     return [_convert_to_message(m) for m in messages]
 
 
-def _runnable_support(func: Callable) -> Callable:
-    @overload
-    def wrapped(
-        messages: None = None, **kwargs: Any
-    ) -> Runnable[Sequence[MessageLikeRepresentation], list[BaseMessage]]: ...
+P = ParamSpec("P")  # Parameters of the decorated function
+R = TypeVar("R")  # Return type of the decorated function
 
-    @overload
-    def wrapped(
-        messages: Sequence[MessageLikeRepresentation], **kwargs: Any
-    ) -> list[BaseMessage]: ...
 
+def _runnable_support(
+    func: Callable[Concatenate[Iterable[MessageLikeRepresentation], P], R],
+) -> Callable[
+    Concatenate[Iterable[MessageLikeRepresentation] | None, P], R | RunnableLambda
+]:
+    # @wraps(func)
     def wrapped(
-        messages: Sequence[MessageLikeRepresentation] | None = None,
-        **kwargs: Any,
-    ) -> (
-        list[BaseMessage]
-        | Runnable[Sequence[MessageLikeRepresentation], list[BaseMessage]]
-    ):
+        messages: Iterable[MessageLikeRepresentation] | None = None,
+        /,
+        *args: P.args,
+        **kwargs: P.kwargs,
+    ) -> R | RunnableLambda:
         # Import locally to prevent circular import.
         from langchain_core.runnables.base import RunnableLambda  # noqa: PLC0415
 
         if messages is not None:
-            return func(messages, **kwargs)
+            return func(messages, *args, **kwargs)
         return RunnableLambda(partial(func, **kwargs), name=func.__name__)
 
     wrapped.__doc__ = func.__doc__
     return wrapped
 
 
-@_runnable_support
+@overload
+def filter_messages(
+    messages: Iterable[MessageLikeRepresentation] | PromptValue,
+    *,
+    include_names: Sequence[str] | None = None,
+    exclude_names: Sequence[str] | None = None,
+    include_types: Sequence[str | type[BaseMessage]] | None = None,
+    exclude_types: Sequence[str | type[BaseMessage]] | None = None,
+    include_ids: Sequence[str] | None = None,
+    exclude_ids: Sequence[str] | None = None,
+    exclude_tool_calls: Sequence[str] | bool | None = None,
+) -> list[BaseMessage]: ...
+
+
+@overload
+def filter_messages(
+    messages: None = None,
+    *,
+    include_names: Sequence[str] | None = None,
+    exclude_names: Sequence[str] | None = None,
+    include_types: Sequence[str | type[BaseMessage]] | None = None,
+    exclude_types: Sequence[str | type[BaseMessage]] | None = None,
+    include_ids: Sequence[str] | None = None,
+    exclude_ids: Sequence[str] | None = None,
+    exclude_tool_calls: Sequence[str] | bool | None = None,
+) -> RunnableLambda[
+    Iterable[MessageLikeRepresentation] | PromptValue, list[BaseMessage]
+]: ...
+
+
+@_runnable_support  # type: ignore[misc]
```

Comment:
The overloads are correct but don't correspond to the impl because of the decorator.